File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-9.js
class Solution {
    /**
     * @param {string[]} tokens
     * @return {number}
     */
    evalRPN(tokens) {
        const stack = []
        const operators = new Set(['+', '-', '*', '/'])

        for (const token of tokens) {
            if (operators.has(token)) {
                const num2 = stack.pop()
                const num1 = stack.pop()
                let tmp = 0

                if (token === '+') {
                    tmp = num1 + num2
                } else if (token === '-') {
                    tmp = num1 - num2
                } else if (token === '*') {
                    tmp = num1 * num2
                } else {
                    tmp = Math.trunc(num1 / num2)
                }
                stack.push(parseInt(tmp))

            } else {
                stack.push(parseInt(token))
            }
        }
        return stack.pop()
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-9.py
class Solution:
    def isValid(self, s: str) -> bool:
        parenthesis = {
            ')': '(',
            ']': '[',
            '}': '{'
        }

        stack = []

        for p in s:
            if stack and p in parenthesis:
                if parenthesis[p] == stack[-1]:
                    stack.pop()
                    continue
                else:
                    return False
            
            stack.append(p)
        
        return len(stack) == 0
